src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -332,6 +327,12 @@ void ClusterTaskManager::TaskFinished(std::shared_ptr<WorkerInterface> worker,
                                       Task *task) {
   RAY_CHECK(worker != nullptr && task != nullptr);
   *task = worker->GetAssignedTask();
+
+  if (!task->GetTaskSpecification().GetDependencies().empty()) {
```

Comment:
What's the side effect of this change? It looks like CancePull happens after this is called. It doesn't have any impact to number of pull/push requests in this case? Just curious.

Reply:
As long as the dependencies are subscribed, they will count towards the objects tracked by the PullManager. So this change delays when we unsubscribe from dependencies so that they will be included in the 70% reservation.

Reply:
Btw can we add a comment here saying the above?